pandas/tseries/index.py

```diff
@@ -832,8 +832,8 @@ def astype(self, dtype):
 
         if dtype == np.object_:
             return self.asobject
-        elif dtype == _INT64_DTYPE:
-            return self.asi8.copy()
+        elif issubclass(np.dtype(dtype).type, np.integer):
```

Comment:
actually, better to use ``is_integer_dtype(dtype)`` more generic her

(and change ``dtype == np.object_`` to ``is_object_dtype(dtype)``)
and ``dtype == _NS_DTYPE`` to ``is_int64_dtype(dtype)``
and ``dtype == str`` to ``is_string_dtype(dtype)``